lighthouse-cli/performance-experiment/server.js

```diff
@@ -54,25 +59,62 @@ function serveAndOpenReport(lighthouseParams, results) {
 }
 
 function requestHandler(request, response) {
-  const pathname = path.normalize(parse(request.url).pathname);
+  request.parsedUrl = parse(request.url, true);
+  const pathname = request.parsedUrl.pathname;
 
-  if (pathname === '/') {
-    reportRequestHandler(request, response);
+  if (request.method === 'GET') {
+    if (pathname === '/') {
+      reportRequestHandler(request, response);
+    } else {
+      response.writeHead(404);
+      response.end('404: Resource Not Found');
+    }
+  } else if (request.method === 'POST') {
+    if (pathname === '/rerun') {
+      rerunRequestHandler(request, response);
+    } else {
+      response.writeHead(404);
+      response.end('404: Resource Not Found');
+    }
   } else {
-    response.writeHead(400);
-    response.write('400 - Bad request');
-    response.end();
+    response.writeHead(405);
+    response.end('405: Method Not Supported');
   }
 }
 
 function reportRequestHandler(request, response) {
-  const reportGenerator = new ReportGenerator();
-  const html = reportGenerator.generateHTML(lhResults, 'cli');
+  const experimentData = hostedExperiments[request.parsedUrl.query.id || 'original'];
+  if (!experimentData) {
+    response.writeHead(404);
+    response.end('404: Resource Not Found');
+    return;
+  }
+
+  const html = reportGenerator.generateHTML(experimentData.results, 'perf-x');
   response.writeHead(200, {'Content-Type': 'text/html'});
-  response.write(html);
-  response.end();
+  response.end(html);
+}
+
+function rerunRequestHandler(request, response) {
+  let message = '';
+  request.on('data', data => message += data);
+
+  request.on('end', () => {
+    const url = hostedExperiments.original.params.url;
+
+    // Add more to flags without changing the original flags
+    const additionalFlags = JSON.parse(message);
+    const flags = Object.assign({}, hostedExperiments.original.params.flags, additionalFlags);
+
+    lighthouse(url, flags, perfOnlyConfig).then(results => {
```

Comment:
Could you explain in a little bit more detail? I don't know how to make it reusable by extension or devtools. Thanks : )